### ISSUE TITLE
BUG: Unexpected changes of the geometry column of GeoDataFrame

### ISSUE THREAD

**Issue by makwingchi** [geopandas/geopandas#2017]
- [ ] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of geopandas.

---

```python
>>> import pandas as pd
>>> import geopandas as gpd

>>> test_dict = {"x1":[1,2], "y1":[1,2], "x2":[3,4], "y2":[3,4]}
>>> test = pd.DataFrame(test_dict)
>>> test["geom"] = gpd.points_from_xy(test["x2"], test["y2"])
>>> test1 = gpd.GeoDataFrame(test, geometry="geom")
```

At this point, test and test1 look like:
```python
>>> test
   x1  y1  x2  y2                     geom
0   1   1   3   3  POINT (3.00000 3.00000)
1   2   2   4   4  POINT (4.00000 4.00000)

>>> test1
   x1  y1  x2  y2                     geom
0   1   1   3   3  POINT (3.00000 3.00000)
1   2   2   4   4  POINT (4.00000 4.00000)
```
which is as expected.

But after I add this line,
```python
test["geom"] = gpd.points_from_xy(test["x1"], test["y1"])
```

it affects the "geom" column of test1 as well.
```python
>>> test
   x1  y1  x2  y2                     geom
0   1   1   3   3  POINT (1.00000 1.00000)
1   2   2   4   4  POINT (2.00000 2.00000)

>>> test1
   x1  y1  x2  y2                     geom
0   1   1   3   3  POINT (1.00000 1.00000)
1   2   2   4   4  POINT (2.00000 2.00000)
```

#### Problem description

The geom column of test1 should not change after I assign new values to the geom column of the origin dataframe test.

#### Expected Output
```python
>>> test["geom"] = gpd.points_from_xy(test["x2"], test["y2"])
>>> test1 = gpd.GeoDataFrame(test, geometry="geom")
>>> test["geom"] = gpd.points_from_xy(test["x1"], test["y1"])
>>> test1
   x1  y1  x2  y2                     geom
0   1   1   3   3  POINT (3.00000 3.00000)
1   2   2   4   4  POINT (4.00000 4.00000)
```

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.8.3 (default, Jul  2 2020, 17:30:36) [MSC v.1916 64 bit (AMD64)]
executable : D:\anaconda3\python.exe
machine    : Windows-10-10.0.19041-SP0

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : None
GEOS lib   : None
GDAL       : 3.2.3
GDAL data dir: None
PROJ       : 7.2.1
PROJ data dir: D:\anaconda3\lib\site-packages\pyproj\proj_dir\share\proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.9.0
pandas     : 1.2.4
fiona      : 1.8.19
numpy      : 1.20.3
shapely    : 1.7.1
rtree      : 0.9.4
pyproj     : 3.0.1
matplotlib : 3.4.2
mapclassify: None
geopy      : None
psycopg2   : None
geoalchemy2: None
pyarrow    : None
pygeos     : None

</details>

**Comment by martinfleis** [comment]
Hi, thanks for the report!

This is a known issue reported as #1179, which _may_ be fixed in the next release if we merge #1324.

**Comment by makwingchi** [comment]
> Hi, thanks for the report!
> 
> This is a known issue reported as #1179, which _may_ be fixed in the next release if we merge #1324.

Thanks Martin, apologies for the duplication!
